### DISCUSSION TITLE
When Argo Helm supports 3.4.0 version of argo workflows

### DISCUSSION THREAD

**Discussion by pockyhe** [argoproj/argo-helm#1406] [Q&A] [answered]
Hi, I ‘m looking forward to the release of 3.4.0 version. We wanna enable Azure Storage Blob feature in k8s by installing helm chart. However, Argo-Helm now just support Argo workflow 3.3.9 which didn't include this new feature. We really hope to know your plan about  when the Argo Helm Chart will include this version. Thank you very much

**Comment by jmeridth** [comment]
@pockyhe great question.

We typically are pretty fast about getting a new helm chart release when a new version releases for one of the argo suite.  Since this is a minor version update (3.4.0), there will most likely be a decent amount of values.yaml/template changes necessary.  I know there is a [pre-release candidate](https://github.com/argoproj/argo-workflows/releases/tag/v3.4.0-rc1) of argo-workflows currently.  @mkilchhofer do we want to entertain starting a helm chart for a release candidate?  Have we done this in the past? Would be proactive for the eventual release and allow us to work through any issues early.

**Reply by mkilchhofer** [reply]
I would not start to maintain charts for pre releases as we are not prepared for release branches (we do also not publish charts for older minor releases with backported fixes). Also it would increase load on reviewers.

I did not analyze the changes on 3.4 but our charts supports overriding the image tag. Are there other changes required @jmeridth?

**Reply by jmeridth** [reply]
@mkilchhofer I'm diving into the changes today.

**Reply by terrytangyuan** [reply]
Here are the missing parts in the current Helm templates on top of my head:
1. A new CRD workflowartifactgctasks: https://github.com/argoproj/argo-workflows/blob/master/manifests/base/crds/full/argoproj.io_workflowartifactgctasks.yaml
2. Permissions on the new CRD workflowartifactgctasks: https://github.com/argoproj/argo-workflows/blob/master/manifests/cluster-install/workflow-controller-rbac/workflow-controller-clusterrole.yaml#L44


Hope this helps.

**Reply by jmeridth** [reply]
@terrytangyuan thank you for that information.  I'm also looking at https://argoproj.github.io/argo-workflows/upgrading/#upgrading-to-v34 to make sure we're not missing anything.

**Reply by terrytangyuan** [reply]
Just realized that we missed some upgrading notes regarding executors. https://github.com/argoproj/argo-workflows/pull/9437

Let us know if we missed anything else.

**Comment by jmeridth** [comment]
Got information that the maintainers of argo-workflows are looking to release 3.4.0 during ArgoCon 2022 (9/19/22-9/21/22).  🤞

**Reply by tooptoop4** [reply]
its released!

**Reply by jmeridth** [reply]
@tooptoop4 saw that.  prepping the helm chart.  someone else may beat me to it.  we'll see.

**Reply by pockyhe** [reply]
good news!!

**Reply by pockyhe** [reply]
I'm very looking forward to enable argo workflow v3.4.0 in argo helm

**Reply by jmeridth** [reply]
@pockyhe Yep, @yu-croco beat me to it.  He's missing the new CRD.  We're coordinating.  Cheers and thank you for your patience.

**Comment by jmeridth** [comment, marked as answer]
This will be complete once [this PR](https://github.com/argoproj/argo-helm/pull/1463) merges

**Comment by jmeridth** [comment]
It has been merged

**Reply by pockyhe** [reply]
Amazing. I found new version of  Helm chart for Argo workflow: 0.18.0. Is it released version of helm chart that enable argo workflow 3.4.0?

**Reply by jmeridth** [reply]
@pockyhe yes
